hiding progress view from super view to allow user interactions

ryanfitz opened the pull request:
This change removes the progress view from its superview after its final animation has completed. This is needed so that users can interact with views that are underneath the progress view.

kishikawakatsumi commented:
Not to remove the view when animation is completed is by design. I assume the use cases that want to start the animation again.

But your suggestion also seems reasonable. So could you please be able to use either by conditions? For example, add `removeOnCompletion` property.

ryanfitz commented:
I didn't realize leaving the view was by design. I changed the implementation to now hide the progress view instead of removing it. The progress view will now reappear if the progress is reset. 

It seemed like hiding both the view as well as its background view was redundant. I removed the backgroundView hide calls and switched to hiding the main view. Let me know if the background view should hide calls should still exist.

kishikawakatsumi commented:
It looks nice. Thank you very much for your contribution. :pray:

kishikawakatsumi commented:
Released v1.0.1. https://github.com/kishikawakatsumi/UCZProgressView/blob/master/CHANGELOG.md
Thanks again.
